nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/probes/ProbeManagerImpl.java

```diff
@@ -218,4 +230,54 @@ protected void doRunProbe(ProbeInfo probe) {
         }
     }
 
+    @Override
+    public void registerProbeForHealthCheck(HealthCheckProbesDescriptor descriptor) {
+        String name = descriptor.getName();
+        if (!descriptor.isEnabled()) {
+            if (probesForHealthCheck.containsKey(name)) {
+                probesForHealthCheck.remove(name);
+                return;
+            }
+        }
+        probesForHealthCheck.put(name, infosByShortcuts.containsKey(name) ? getProbeInfo(name) : null);
+    }
+
+    @Override
+    public Collection<ProbeInfo> getAllContributeToHealthCheckProbeInfos() {
+        return Collections.unmodifiableCollection(probesForHealthCheck.values());
+    }
+
+    @Override
+    public HealthCheckResult getOrRunHealthCheck() {
+        for (String probeName : probesForHealthCheck.keySet()) {
+            ProbeInfo probe = probesForHealthCheck.get(probeName);
```

Comment:
for (EntrySet<String, ProveInfo> es : probesForHealthCheck.entrySet()) {
            String probeName = es.getKey();
            ProbeInfo probe = es.getValue();